python/tests/test_conftest.py

```diff
@@ -429,3 +445,37 @@ def loop(b):
             res = pytester.runpytest()
             res.ret == 0
             pytest.raises(TimeoutException)
+
+    def test_check_order_tests(self, pytester: Pytester):
+        checkfile = pytester.makepyfile(
+            """ 
+        from inline import Here
+        def m(a):
+            a = a + 1
+            Here(tag = ["add"]).given(a, 1).check_eq(a, 2)
+            a = a - 1
+            Here(tag = ["minus"]).given(a, 1).check_eq(a, 0)
+    """
+        )
+        for x in (pytester.path, checkfile):
+            reprec = pytester.inline_run("--inlinetest-order=add")
+            items = [x.item for x in reprec.getcalls("pytest_itemcollected")]
+            assert len(items) == 2
```

Comment:
This assertion may be a bit weak, because it only checks the number of inline tests but not the order of them. Maybe you can give each inline test a name and write a stronger assertion:
```
Here("1", tag = ["add"])...
Here("2", tag = ["minus"])...

assert len(items) == 2
assert items[0].test_name == "1"
assert items[1].test_name == "2"
```

Reply:
also I feel like you should add a test that actually changes the default order, to make sure the ordering algorithm is functioning. For example, with the same code but `pytester.inline_run("--inlinetest-order=minus")`